src/providers/gitlab.js

```diff
@@ -0,0 +1,28 @@
+const gitlabConfig = {
+  name: 'gitlab',
+  selectors: {
+    // Row in file list, file view header
+    row: 'table[data-qa-selector="file_tree_table"] tr, .file-header-content',
+    // Cell in file list, file view header, readme header
+    filename: 'td.tree-item-file-name, .file-title-name, .gl-link',
+    // Any icon not contained in a button
+    icon: 'svg:not(.gl-button-icon)',
+  },
+  getIsLightTheme: () => !document.querySelector('body').classList.contains('gl-dark'),
+  getIsDirectory: ({ icon }) => icon.getAttribute('data-testid') === 'folder-icon',
+  getIsSubmodule: ({ row }) => row.querySelector('a')?.classList.contains('is-submodule') || false,
```

Comment:
Do we need the `|| false` here? If `contains` always returns a boolean we'll always ignore that because `(true || false) === true` and `(false || false) === false`.
```suggestion
  getIsSubmodule: ({ row }) => row.querySelector('a')?.classList.contains('is-submodule'),
```

Reply:
The reason for this is that in the case of the file header icons, this element won't get selected (hence the optional chaining), and in that case this would return `undefined`.  And while that wouldn't actually matter, as `undefined` will still count as falsey when the value is checked, my TypeScript brain thought it would be better to keep it consistently returning a boolean haha.  Idk what your thoughts are on that.

Reply:
hahah ok, that's fair. Yeah, no particular preference here then. I always just read the result of this method as `contains` or `not contains` but I understand it could be cleaner to enforce a boolean return. lgtm then